x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/ScalarFunction.java

```diff
@@ -68,6 +69,9 @@ protected ScriptTemplate asScript(Expression exp) {
             if (attr instanceof AggregateFunctionAttribute) {
                 return asScriptFrom((AggregateFunctionAttribute) attr);
             }
+            if (attr instanceof LiteralAttribute) {
```

Comment:
Because at that point the literal expression is an `Alias` which is not foldable and then becomes a `LiteralAttribute` from the `Expressions.attribute(exp)` call.